app.rb

```diff
@@ -0,0 +1,88 @@
+require './person'
+require './student'
+require './teacher'
+require './books'
+require './rental'
+require './classroom'
+
+class Apps
+  def initialize
+    @people = []
+    @books = []
+    @rentals = []
+  end
+
+  def list_book
+    if @books.empty?
+      puts 'No books available'
+    else
+      @books.each { |book| puts "Title: #{book.title}, Author: #{book.author}" }
+    end
+  end
+
+  def list_person
+    if @people.empty?
+      puts 'No one is available'
+    else
+      @people.each do |person|
+        puts "[#{person.class}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
+      end
+    end
+  end
+
+  def create_person
+    print 'Do you want to create a student (1) or a teacher (2)? [Input the number]: '
+    person_type = gets.chomp
+    if person_type != '1' && person_type != '2'
+      puts 'Invalid option'
+      return
+    end
+
+    print 'Age: '
+    age = gets.chomp
+
+    print 'Name: '
+    name = gets.chomp
+
+    person = person_type == '1' ? Student.new(age, name) : Teacher.new(age, name)
+    puts 'Person created successfully'
+
+    @people << person
+  end
+
+  def create_book
+    print 'Title: '
+    title = gets.chomp
+
+    print 'Author: '
+    author = gets.chomp
+
+    @books << Book.new(title, author)
+    puts 'Book created successfully'
+  end
+
+  def create_rental
+    print 'Select a book from the following list by number'
+    @books.each_with_index { |book, index| puts "#{index}) Title: #{book.title}, Author: #{book.author}" }
+    book_index = gets.chomp.to_i
+    print 'Select a person from the following list by number'
+    @people.each_with_index do |person, index|
+      puts "#{index}) [#{person.class}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
+    end
+    person_index = gets.chomp.to_i
+    print 'Date: '
+    date = gets.chomp
+
+    @rentals << Rental.new(date, @books[book_index], @people[person_index])
+    puts 'Rental created successfully'
+  end
+
+  def list_rentals_for_person_id
+    print 'ID of person: '
+    id = gets.chomp
+
+    rentals = @rentals.filter { |rental| rental.person.id == id }
+    puts 'Rentals:'
+    rentals.each { |rental| puts "Date: #{rental.date}, Book: #{rental.book.title} by #{rental.book.author}" }
+  end
```

Comment:
- [ ] I'm not able to list the rentals because of the error below. Kindly fix that.

<img width="594" alt="image" src="https://user-images.githubusercontent.com/77190203/194584680-f94247ba-ac2f-4b85-8ecf-761f467f81af.png">

Reply:
Thank you I will fix it